### ISSUE TITLE
Incomplete DAG with checkpoints

### ISSUE THREAD

**Issue by fgvieira** [snakemake/snakemake#2822]
**Snakemake version**

`v8.10.7`

**Describe the bug**

DAG is incomplete when there are checkpoints, since it only plots the DAG for the rules upstream of the checkpoint.

**Minimal example**

On test `snakemake/tests/test_checkpoint_allowed_rules`, if we run `snakemake c --dag | dot -Tsvg > dag.svg`, we get the DAG:

![dag](https://github.com/snakemake/snakemake/assets/1151762/38317477-0667-4099-8e41-5bd7ba22340e)

Where I think we would also expect rule `b` going into rule `c`.

**Comment by cademirch** [comment]
Seems slightly related to #2824. I think this makes sense since B doesn't exist in the DAG until after A has been run.

**Comment by fgvieira** [comment]
Yes, but then it should work if `a` has been run, but it does not. As it is now:
- if `a` has not been run: `a -> c`
- if `a` has bee run: `b -> c`

So it is never possible to get the full DAG.

**Comment by bsmith89** [comment]
Is this also related to https://github.com/snakemake/snakemake/issues/2233?

**Comment by fgvieira** [comment]
Could very well be! 😄

**Comment by johanneskoester** [comment]
Good catch!

**Comment by KewinOgink** [comment]
Workaround is that you run the pipeline and produce data until the checkpoint, and then use --dag.
